_posts/2021-03-14-Julia-1.6-what-has-changed-since-1.0.md

```diff
@@ -0,0 +1,737 @@
+
+This is an experimental features, so it's not covered by the SemVer guarantees of the rest of the language.
+Still it's neat, and I don't expect it to go away,
+though I also don't expect it to graduate from experimental status until a bunch of people are using it.
+Which they will probably do as people start dropping support for older Julia versions.
+
+
+### A bunch of curried functions
+Julia has have ended up with a convention of providing curried methods of functions if that would be useful as the first argument for `filter`.
+For example, `filter(isequal(2), [1,2,3,2,1])` is the same is `filter(x->isequal(x, 2), [1,2,3,2,1])`.
+In particular these are boolean comparison-like functions:
+with two arguments, where the thing being compared against is the second.
+Julia 1.0 had `isequal`, `==` and `in`.
+Since then we have added:
+`isapprox`, `<`,`<=`,`>`, `>=`, `!=`, `startswith`, `endswith`, and `contains`.
+
+Aside: `contains` is argument-flipped `occursin`.
+It was a thing in Julia 0.6, but was removed in 1.0 and now has been added back.
+We added it back primarily so we could have the curried form, and to match `startswith` and `endswith`.
+
+### A ton of other new and improved standard library functions
+
+I am not going to manage to list all of them here.
+But I will list some of my standout favorites.
+
+
+[`@time`](https://github.com/JuliaLang/julia/pull/37678) now reports how much time was spent on compilation.
+This is going to help prevent people new to the language from including compilation time in their benchmarks.
+It is still better to use [BenchmarkTools.jl's](https://github.com/JuliaCI/BenchmarkTools.jl) `@btime`, since that does multiple samples.
+But now that too can report time spent on compilation.
+It's also useful for identifying if certain functions are taking ages to compile.
+Which I guess is its theoretical main point, but I think preventing people from benchmarking wrong is going to come up way more often.
+
+The experimental [`Base.@locals`](https://github.com/JuliaLang/julia/issues/29733)
+was added, which returns a dictionary of local variables.
+That one surprised me; I though being able to access a dictionary of local variables would get in the way of the optimizer, since it would prevent it from being able to optimize variables that are used for intermediate values away entirely.
+But the compiler folks know better than I do.
+
+[`splitpath`](https://github.com/JuliaLang/julia/issues/28156) is added, it's the opposite of `joinpath`.
+Kind of silly we didn't have that, and had been bugging me at least since 0.6.
+
+On things that had been bugging me, I had wanted [`eachslice` and its special cases: `eachrow` & `eachcol`](https://github.com/JuliaLang/julia/issues/29749) since Julia 0.3 when I first started using it.
```

Comment:
One thing that bugged me - I wanted ...

Reply:
```suggestion
Similarly, I have wanted [`eachslice` and its special cases: `eachrow` & `eachcol`](https://github.com/JuliaLang/julia/issues/29749) since Julia 0.3 when I first started using the language.
```